src/Fmt_ast.ml

```diff
@@ -2654,6 +2669,45 @@ and fmt_class_type_field c ctx (cf : class_type_field) =
     $ fmt_atrs )
 
 and fmt_cases c ctx cs =
+  let rec pattern_len ?(parens = 0) pat =
+    match pat.ppat_desc with
+    | Ppat_any -> Some 1
+    | Ppat_var {txt= s; _}
+     |Ppat_constant (Pconst_integer (s, _))
+     |Ppat_constant (Pconst_float (s, _))
+     |Ppat_construct ({txt= Lident s; _}, None) ->
+        Some (String.length s)
+    | Ppat_construct ({txt= Lident s; _}, Some arg) -> (
+      match pattern_len ~parens:2 arg with
+      | Some arg -> Some (parens + String.length s + 1 + arg)
+      | None -> None )
+    | Ppat_constant (Pconst_char _) -> Some 4 (* when the char is escaped *)
```

Comment:
We can't make approximation:

```ocaml
let _ =
  match f with
  | 'a'    -> 1
  | '\n'    -> 2
  | '\x12'    -> 2
  | pattern -> 3
```

This depends on concrete syntax and `escape-chars` option. I suggest making `fmt_char_escaped` return a string instead and using it to measure what will be printed.

Reply:
Fixed.